kubeflow/pipelines.py

```diff
@@ -0,0 +1,445 @@
+    ),
+    config: str = "datasetinsights/configs/faster_rcnn_groceries_real.yaml",
+    checkpoint_file: str = (
+        "https://storage.googleapis.com/datasetinsights/models/"
+        "fine-tuned-sim2real/FasterRCNN.estimator"
+    ),
+    tb_log_dir: str = "gs://<bucket>/runs/yyyymmdd-hhmm",
+    volume_size: str = "100Gi",
+):
+    output = test_data = DATA_PATH
+
+    # The following parameters can't be `PipelineParam` due to this issue:
+    # https://github.com/kubeflow/pipelines/issues/1956
+    # Instead, they have to be configured when the pipeline is compiled.
+    memory_limit = "64Gi"
+    num_gpu = 1
+    gpu_type = "nvidia-tesla-v100"
+
+    # Pipeline definition
+    vop = volume_op(volume_size=volume_size)
+    download = download_op(
+        docker=docker,
+        source_uri=source_uri,
+        output=output,
+        volume=vop.volume,
+        memory_limit=memory_limit,
+    )
+    evaluate = evaluate_op(
+        docker=docker,
+        config=config,
+        checkpoint_file=checkpoint_file,
+        test_data=test_data,
+        tb_log_dir=tb_log_dir,
+        volume=download.pvolumes[DATA_PATH],
+        memory_limit=memory_limit,
+        num_gpu=num_gpu,
+        gpu_type=gpu_type,
+    )
+
+    return evaluate
+
+
+@dsl.pipeline(
+    name="Train on real world dataset",
+    description="Train on real world dataset",
+)
+def train_on_real_world_dataset(
+    docker: str = "unitytechnologies/datasetinsights:latest",
+    source_uri: str = (
+        "https://storage.googleapis.com/datasetinsights/data/groceries/v3.zip"
+    ),
+    config: str = "datasetinsights/configs/faster_rcnn_groceries_real.yaml",
+    tb_log_dir: str = "gs://<bucket>/runs/yyyymmdd-hhmm",
+    checkpoint_dir: str = "gs://<bucket>/checkpoints/yyyymmdd-hhmm",
+    volume_size: str = "100Gi",
+):
+    output = train_data = val_data = DATA_PATH
+
+    # The following parameters can't be `PipelineParam` due to this issue:
+    # https://github.com/kubeflow/pipelines/issues/1956
+    # Instead, they have to be configured when the pipeline is compiled.
+    memory_limit = "64Gi"
+    num_gpu = 1
+    gpu_type = "nvidia-tesla-v100"
+
+    # Pipeline definition
+    vop = volume_op(volume_size=volume_size)
+    download = download_op(
+        docker=docker,
+        source_uri=source_uri,
+        output=output,
+        volume=vop.volume,
+        memory_limit=memory_limit,
+    )
+    train = train_op(
+        docker=docker,
+        config=config,
+        train_data=train_data,
+        val_data=val_data,
+        tb_log_dir=tb_log_dir,
+        checkpoint_dir=checkpoint_dir,
+        volume=download.pvolumes[DATA_PATH],
+        memory_limit=memory_limit,
+        num_gpu=num_gpu,
+        gpu_type=gpu_type,
+    )
+
+    return train
+
+
+@dsl.pipeline(
+    name="Train on Synthetic + Real World Dataset",
+    description="Train on Synthetic + Real World Dataset",
+)
+def train_on_synthetic_and_real_dataset(
```

Comment:
Should this pipeline include downloading of the synthetic dataset? Currently it only downloads the groceries real dataset from what it looks like.

Reply:
This pipeline actually does not require Synthetic Dataset. It should consume a model train on a dif

Reply:
The function name is a bit misleading then. But its a minor thing. If the model is doing fine tuning then it should say so i think.